src/components/decorators.ts

```diff
@@ -1,40 +1,75 @@
-type FieldType<T> = T extends string
-  ? StringConstructor
-  : T extends number
-    ? NumberConstructor
-    : T extends boolean
-      ? BooleanConstructor
-      : T extends unknown[]
-        ? ArrayConstructor
-        : never
-
-type ConstructorMetadata<T extends HTMLElement> = {
-  new (): T
-  properties?: { [K in keyof T]?: FieldType<T[K]> }
+type ConstructorWithProperties = {
+  new (...args: unknown[]): HTMLElement
   observedAttributes?: string[]
 }
 
-type Flags = {
-  observe?: boolean
+type PropertyOptions = {
```

Comment:
simplify the PropertyOptions type to:
type PropertyOptions = StringConstructor | NumberConstructor | BooleanConstructor | typeof Array